### PR TITLE
PYTHON-4962 Adopt zizmor GitHub Actions security scanner

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
I opened https://jira.mongodb.org/browse/MOTOR-1413 to track the failures.

**Review by NoahStapp — APPROVED** [review]

File: .github/workflows/zizmor.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: GitHub Actions Security Analysis with zizmor 🌈
```
Comment (NoahStapp) [inline .github/workflows/zizmor.yml:1]: Is this 🌈 emoji actually part of the name?

File: .github/workflows/zizmor.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Setup Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Get zizmor
+        run: cargo install zizmor
+      - name: Run zizmor 🌈
```
Comment (NoahStapp) [inline .github/workflows/zizmor.yml:24]: Same here.

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .github/workflows/zizmor.yml (hunk 1 above)
Comment (blink1073) [reply .github/workflows/zizmor.yml:1]: No it was part of the example in the zizmore repo

Comment (NoahStapp) [reply .github/workflows/zizmor.yml:1]: Then can we remove it? It's a little distracting to have emojis in config files.

Comment (blink1073) [reply .github/workflows/zizmor.yml]: Done

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**